src/data/list/basic.lean

```diff
@@ -1222,6 +1222,28 @@ ext $ λn, if h₁ : n < length l₁
   then by rw [nth_le_nth, nth_le_nth, h n h₁ (by rwa [← hl])]
   else let h₁ := le_of_not_gt h₁ in by { rw [nth_len_le h₁, nth_len_le], rwa [←hl], }
 
+lemma injective_append_left (l : list α) : function.injective ((++) l) :=
+begin
+  induction l with hd tl IH,
+  { convert function.injective_id },
+  { intros as bs h,
+    refine IH _,
+    simpa using h }
+end
+
+lemma injective_append_right (l : list α) : function.injective (++ l) :=
+begin
+  intros as bs h,
+  simp only at h,
+  have : as.length = bs.length := by simpa using congr_arg length h,
+  apply ext_le this,
+  intros n hn hn',
+  have ha : n < (as ++ l).length := hn.trans_le (by simp),
+  have hb : n < (bs ++ l).length := hn'.trans_le (by simp),
+  simp [←nth_le_append ha, ←nth_le_append hb, h]
+end
+
+
```

Comment:
```suggestion
```
Not only do you not use these lemmas, but they already exist as `append_left_injective` and `append_right_injective`.